src/api/app/models/concerns/staging_project.rb

```diff
@@ -197,13 +197,15 @@ def add_broken_packages(result, building)
     result.elements('status') do |status|
       code = status.get('code')
 
-      if code.in?(['broken', 'failed']) || (code == 'unresolvable' && !building)
-        @broken_packages << { package: status['package'],
-                              project: name,
-                              state: code,
-                              details: status['details'],
-                              repository: result['repository'],
-                              arch: result['arch'] }
+      unless @broken_packages.map { |p| p[:package] }.include?(status['package'])
```

Comment:
Instead of always mapping the array into another array, how about 

```suggestion
      unless @broken_packages.none? { |p| p[:package] == status['package'] }
```

Reply:
@coolo much better, thanks!